File: LeetCode_ProblemSet/Problem590.py
"""
# Definition for a Node.
class Node(object):
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution(object):
    def postorder(self, root):
        """
        :type root: Node
        :rtype: List[int]
        """
        nodes = []
        self.postorderHelp(root, nodes)
        return nodes
        
        
    def postorderHelp(self, root, nodes):
        if root is None:
            return 

        for child in root.children:
            self.postorderHelp(child, nodes)
            
        nodes.append(root.val)

File: LeetCode_ProblemSet/Go/Problem590.go
package main

/**
 * Definition for a Node.
 * type Node struct {
 *     Val int
 *     Children []*Node
 * }
 */

func postorder(root *Node) []int {
	nodes := []int{}
	postorderHelp(root, &nodes)
	return nodes
}

func postorderHelp(root *Node, nodes *[]int) {
	if root == nil {
		return 
	}
	for _, child := range root.Children {
		postorderHelp(child, nodes)
	}
	*nodes = append(*nodes, root.Val)
}